File: backend/Expense_tracker_service/src/main/java/com/monexel/expensetracker/service/BorrowedMoneyService.java
package com.monexel.expensetracker.service;


import java.time.LocalDate;
import java.util.List;

import com.monexel.expensetracker.request.BorrowedMoneyRequest;
import com.monexel.expensetracker.response.BorrowedMoneyResponse;

public interface BorrowedMoneyService {

	BorrowedMoneyResponse addBorrowedMoney(BorrowedMoneyRequest request);
    BorrowedMoneyResponse updateBorrowedMoney(Long id, BorrowedMoneyRequest request);
    void deleteBorrowedMoney(Long id);
    BorrowedMoneyResponse getBorrowedMoneyById(Long id);
    List<BorrowedMoneyResponse> getAllBorrowedMoneyByUser(Long userId,LocalDate startDate, LocalDate endDate);
   

}

File: frontend/src/services/borrowedMoneyService.js
import api from "./api";
 
export const addBorrowedMoney = async (data) => {
  const response = await api.post("/api/borrowed-money/addBorrowedMoney", data);
  return response.data;
};
 
export const updateBorrowedMoney = async (id, data) => {
  const response = await api.put(`/api/borrowed-money/updateBorrowedMoney/${id}`, data);
  return response.data;
};
 
export const deleteBorrowedMoney = async (id) => {
  await api.delete(`/api/borrowed-money/deleteBorrowedMoney/${id}`);
};
 
export const getBorrowedMoneyById = async (id) => {
  const response = await api.get(`/api/borrowed-money/getBorrowedMoneyById/${id}`);
  return response.data;
};
 
export const getAllBorrowedMoneyByUser = async (userId, startDate, endDate) => {
  const response = await api.get(`/api/borrowed-money/getAllBorrowedMoneyByUser/${userId}?startDate=${startDate}&endDate=${endDate}`);
  return response.data;
};